src/Project.hs

```diff
@@ -78,16 +91,23 @@ import Data.Text.Lazy.Builder as Text
 import qualified Data.Aeson as Aeson
 import qualified Data.Aeson.Encode.Pretty as Aeson
 import qualified Data.IntMap.Strict as IntMap
+import qualified Data.Map.Strict as Map
 
 import Types (PullRequestId (..), Username)
 
+data OutstandingChecks
+  = AnyCheck BuildStatus
+  | SpecificChecks (Map Check BuildStatus)
+  deriving (Eq, Show, Generic)
+  deriving anyclass (FromJSON, ToJSON)
```

Comment:
Can you add some doc comments here (e.g. explaining how this relates to the mandatory checks feature)?